subprojects/core-api/src/main/java/org/gradle/api/artifacts/DirectDependency.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts;
+
+
+/**
+ * Implementations of this interface always describe direct dependencies and not
+ * constraints may optionally considered as dependency.
+ *
+ * @since 4.5
+ */
+public interface DirectDependency extends Dependency {
```

Comment:
Yes. I initially thought that, since we insert this into a stable hierarchy, we should skip `@Incubating`. But we might still decide to move this interface in the hierarchy (or to rename it) without breaking compatibility with anything else in the hierarchy.

I added the annotation and adjusted the _accepted public api changes_.